### DISCUSSION TITLE
Programatically get markdown from a plugin

### DISCUSSION THREAD

**Discussion by JUNIORCO** [facebook/docusaurus#9503] [Q&A]
I'm trying to create a plugin in Docusaurus that can read the user's markdown files programatically and ingest them for data processing.

Is this possible? I don't think any of the lifecycle APIs will work but I could be wrong.

**Comment by ilg-ul** [comment]
> I don't think any of the lifecycle APIs will work but I could be wrong.

Is this the answer? So it is not possible?

**Reply by JUNIORCO** [reply]
I'm not sure hence why I made this discussion, to discuss...

**Reply by slorber** [reply]
I don't know what to answer here. 

All our core content plugins read markdown files programmatically to ingest them for data processing, so technically you have 3 concrete examples? 🤔

**Reply by ilg-ul** [reply]
Good point! I'll try to understand how you do it in the content plug-ins. A link to such a piece of code would help.

**Comment by Josh-Cena** [comment]
You use `loadContent` to query the Markdown files you need and `contentLoaded` to parse them and emit whatever JSON data you need on the client side :) If you want to parse the Markdown server-side, bring your own parser because the core doesn't have a parser built-in; if you want to render it client-side, then just emit a `@site/*` path and make client code read that data using `usePluginData` and then `import` from that path, and MDX loader will take care of the parsing and transform.
